### PR TITLE
Add avifIOStats.quality and qualityAlpha

### PR DESCRIPTION
**Pull request by y-guyon**
Move avifQualityToQuantizer to internal.h.
Add avifQuantizerToQuality() to internal.h.
Add avifqualitytest.cc.

Fix ioStats.colorOBUSize and ioStats.alphaOBUSize at decoding.
Add avifiostatstest.cc.

### CODE REVIEW

File: tests/gtest/avifiostatstest.cc [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  }
+
+  testutil::AvifEncoderPtr encoder(avifEncoderCreate(), avifEncoderDestroy);
+  AVIF_CHECK(encoder != nullptr);
+  encoder->speed = AVIF_SPEED_FASTEST;
+  encoder->quality = quality;
+  encoder->qualityAlpha = qualityAlpha;
+  for (int frame = 0; frame < num_frames; ++frame) {
+    if (encode_as_grid) {
+      const avifCropRect left_rect{0, 0, image->width / 2, image->height};
```
Comment (jzern) [inline tests/gtest/avifiostatstest.cc:39]: assignment syntax (= {0, ...}) is generally preferred for cases like this:
https://abseil.io/tips/88

File: tests/gtest/avifiostatstest.cc [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                                       avifImageDestroy);
+      AVIF_CHECK(avifImageSetViewRect(left_cell.get(), image.get(),
+                                      &left_rect) == AVIF_RESULT_OK);
+      const avifCropRect right_rect{image->width / 2, 0, image->width / 2,
+                                    image->height};
+      testutil::AvifImagePtr right_cell(avifImageCreateEmpty(),
+                                        avifImageDestroy);
+      AVIF_CHECK(avifImageSetViewRect(right_cell.get(), image.get(),
+                                      &right_rect) == AVIF_RESULT_OK);
+      const std::array<avifImage*, 2> pointers{left_cell.get(),
```
Comment (jzern) [inline tests/gtest/avifiostatstest.cc:50]: same here and in other places

File: tests/gtest/avifiostatstest.cc [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+//------------------------------------------------------------------------------
+
+}  // namespace
+}  // namespace libavif
+
+int main(int argc, char** argv) {
+  ::testing::InitGoogleTest(&argc, argv);
+  if (argc != 2) {
+    std::cerr << "There must be exactly one argument containing the path to "
```
Comment (jzern) [inline tests/gtest/avifiostatstest.cc:177]: this file should include <iostream>

**Comment by y-guyon** [comment]
Since `targetSize` is no longer planned to be part of the API (see https://github.com/AOMediaCodec/libavif/pull/1278#issuecomment-1435008724), I moved parts of this PR to #1290.

**Pull request closed without merging**